rust/lance/src/dataset.rs

```diff
@@ -1457,7 +1460,8 @@ impl Dataset {
     /// underlying storage. In order to remove the data, you must subsequently
     /// call [optimize::compact_files()] to rewrite the data without the removed columns and
     /// then call [cleanup::cleanup_old_versions()] to remove the old files.
-    pub async fn drop_columns(&mut self, columns: &[&str]) -> Result<()> {
+    /// Returns: The dataset version number after this write operation completes successfully.
+    pub async fn drop_columns(&mut self, columns: &[&str]) -> Result<u64> {
```

Comment:
These functions all take `&mut self` and the manifest of the dataset should be updated after the call.  So I should be able to get the version like this:

```
let mut dataset = ...
dataset.drop_columns(...);
let version = dataset.version();
```

This way we leave the return value open (some other methods today return statistics about the operation itself, like update or cleanup) but the version is still accessible in a consistent way?

It's possible I am missing something though.